src/lib.rs

```diff
@@ -81,6 +81,9 @@ pub enum ParseError {
     AmPmWithoutHour,
     InvalidHour,
     TimezoneUnsupported,
+    InvalidMinute,
```

Comment:
I'd rather unify the `Invalid*` errors under an `ImpossibleTimestamp` or similarly named error - we keep on running into enough of these, I'd rather not have to maintain so many different but essentially similar error types.

Eventually the `ParseInternalError` and `ParseError` will need to be collapsed, but for right now, would you mind unifying these?

Reply:
What do you have in mind for `ImpossibleTimestamp`? I'd like to use an enum for example:

```rust
pub enum ImpossibleTimestamp {
    InvalidHour,
    InvalidMinute,
    // etc.
}
```

Reply:
My original thought was to turn all usages of `InvalidHour`, `InvalidMonth`, etc. to `ImpossibleTimestamp`. If you want to maintain the specific error conditions I'd potentially be OK with that, but I definitely wouldn't want to make yet another enum.

In the Python version you only get a `ValueError` instead of specific "this field is wrong", which is why I'm trying to collapse as many things as possible.

So, I guess - do you really want to break out an `Error` for each component of the timestamp that could be wrong? If so, I can accept this (pending the failing test) but my initial preference would be `s/InvalidHour/ImpossibleTimestamp/g`.

Reply:
I am OK with a single `ImpossibleTimestamp` variant, but it would be nice to have a error description, would you accept an `ImpossibleTimestamp(String)`? (Or maybe `ImpossibleTimestamp(&'static str)`)

Reply:
Yeah, I think that's a great compromise. My guess is that `&'static str` is preferred because these will be constants that can be embedded in the `.DATA` section, but `String` is fine too.

Reply:
It's done.